test-clients/src/main/java/com/hedera/services/yahcli/suites/BalanceSuite.java

```diff
@@ -25,19 +25,61 @@
 import org.apache.logging.log4j.LogManager;
 import org.apache.logging.log4j.Logger;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
-import static com.hedera.services.bdd.spec.transactions.TxnVerbs.cryptoTransfer;
-import static com.hedera.services.bdd.spec.transactions.TxnVerbs.tokenAssociate;
+import static com.hedera.services.bdd.spec.queries.QueryVerbs.getAccountBalance;
+import static com.hedera.services.bdd.spec.transactions.TxnUtils.isIdLiteral;
 
 public class BalanceSuite extends HapiApiSuite {
 	private static final Logger log = LogManager.getLogger(BalanceSuite.class);
 
+	private final Map<String, String> specConfig;
+	private final List<String> accounts;
+
+	public BalanceSuite(final Map<String, String> specConfig, final String[] accounts) {
+		this.specConfig = specConfig;
+		this.accounts = rationalized(accounts);
+	}
+
+	private List<String> rationalized(final String[] accounts) {
+		return Arrays.stream(accounts)
+				.map(a -> getAccount(a))
+				.collect(Collectors.toList());
+	}
+
+	private String getAccount(final String account) {
+		if(isIdLiteral(account)) {
+			return account;
+		} else {
+			try {
+				int number = Integer.parseInt(account);
```

Comment:
should use `long` instead because shard, realm and account are all `int64` in protobuf.